src/multirust/telemetry.rs

```diff
@@ -0,0 +1,34 @@
+use time;
+use config::Cfg;
+use multirust_utils::utils;
+use rustc_serialize::json;
+
+#[derive(Debug, PartialEq)]
+pub enum TelemetryMode {
+    On,
+    Off,
+}
+
+#[derive(RustcDecodable, RustcEncodable, Debug)]
+pub enum TelemetryEvent {
+    RustcRun { duration_ms: u64, exit_code: i32, errors: Option<String> },
+    ToolchainUpdate { toolchain: String, success: bool } ,
+    TargetAdd { toolchain: String, target: String, success: bool },
+}
+
+#[derive(RustcDecodable, RustcEncodable, Debug)]
+struct LogMessage {
+    log_time_s: i64,
+    event: TelemetryEvent,
+}
+
+pub fn log_telemetry(event: TelemetryEvent, cfg: &Cfg) {
+    let ln = LogMessage { log_time_s: time::get_time().sec, event: event };
+
+    let json = json::encode(&ln).unwrap();
+
+    let now = time::now_utc();
+    let filename = format!("telemetry-{}-{:02}-{:02}", now.tm_year + 1900, now.tm_mon, now.tm_mday);
+
+    let _ = utils::append_file("telemetry", &cfg.multirust_dir.join(&filename), &json);
```

Comment:
Instead of putting these directly in the multirust directory can we emit the logs to `.multirust/telemetry/log-$date.json` or something like that. Keep the main directory uncluttered.

This will require putting the settings file somewhere else. The question of what to do with the telemetry settings is tricky because when running proxies we have to load that setting *fast*, but there's also a good chance that telemetry will have more configuration knobs in the future.

I *suspect* that deserializing a telemetry file every time we run rustc would be prohibitive for performance - in the common case we want to do as little deserialization of configuration as possible. Perhaps instead of storing "on" and "off" in `~/.multirust/telemetry` we can instead just use the presence or absence of a file as the flag indicating that telemetry is on. In other words, what if we had an empty file called `~/.multirust/telemetry-on`, whose presence alone indicates that telemetry is on?